src/cascadia/TerminalControl/UiaTextRange.cpp

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "UiaTextRange.hpp"
+#include "screenInfoUiaProvider.hpp"
+
+using namespace Microsoft::Terminal;
+
+std::deque<UiaTextRange*> UiaTextRange::GetSelectionRanges(_In_ Microsoft::Console::Render::IRenderData* pData,
+                                                           _In_ IRawElementProviderSimple* pProvider)
+{
+    std::deque<UiaTextRange*> ranges;

```

Comment:
Yeah. This is a weird function because it's called "GetSelectionRanges" but it's actually creating all of the UTRs and passing them up. Each UTR is created with a refcount of 1.

We're passed up to...
--> `ScreenInfoUiaProvider::GetSelectionRangeUTRs()` (handed off)
--> `ScreenInfoUiaProviderBase::GetSelection()` (handed off)
--> `TermControlAutomationPeer::GetSelection()` (unwrap/wrap to owning vector)

I THINK this is done properly but I'll default to you since you have more experience in this area.

Reply:
Turned into wil::com_ptr so no worries.